### DISCUSSION TITLE
Force ViewAction to open in a Modal

### DISCUSSION THREAD

**Discussion by gregoryloichot** [filamentphp/filament#14702] [Help]
### Package

Table builder

### Package Version

v3.2.122

### How can we help you?

Hello, 

Is it possible to force a `ViewAction` (in the actions of a table) to open the "detail" view in a modal even if the Resource has a `ViewPage`?

Thanks for your help.

**Comment by leandrocfe** [comment]
Just removing the 'view' page in the `getPages()` method?

**Reply by gregoryloichot** [reply]
Hi, thanks for your answer.

The table is reused on many pages and I want, depending on which page it is, to open a modal or the ViewPage.

**Reply by JBou** [reply]
I had a similar problem, I wanted to open the create action in a `ListRecords` to open in a model, instead of a new page.

```php
class ListYears extends ListRecords
{
    protected static string $resource = YearResource::class;

    protected function getHeaderActions(): array
    {
        return [
            Actions\CreateAction::make()->url(null),
        ];
    }
}
```

I tested it in a table too, you can just append `->url(null)` to the `ViewAction` and it will open in a table:

```php
class UserResource extends Resource
{
    public static function table(Table $table): Table
    {
        return $table
            ->columns([
                //...
            ])
            ->actions([
                EditAction::make(),
                ViewAction::make()->url(null),
            ]);
    }
}
```

It would be useful to document this or add a method to force it (`forceOpenInModal`).
In my case, I want to open it in a Modal, because it is a really simple resource and you immediately see the new item in the list, makes the user experience better. I still need the separate `create` page and route, because I have some buttons that link to the page to create new records.
